src/lib/structures/uid.mli

```diff
@@ -30,10 +30,22 @@ module DE = Dolmen.Std.Expr
 type t = private
   | Hstring : Hstring.t -> t
   | Dolmen : 'a DE.id -> t
+  | Term_cst : DE.term_cst -> t
+  | Ty_cst : DE.ty_cst -> t
```

Comment:
This is orthogonal to the PR but if we define

```ocaml
type _ t =
  | Hstring : Hstring.t -> 'a t
  | Term_cst : DE.term_cst -> DE.ty t
  | Ty_cst : DE.ty_cst -> DE.type_fun t
  | Ty_var : DE.ty_var -> DE.type_ t

type term_cst = DE.ty t

type ty_var = DE.type_ t

type ty_cst = DE.type_fun t
```

then we can expose the "kind" of uid in the type and gain type safety (in particular, `perfect_hash` can be of type `Uid.term_cst -> int`).

Reply:
I applied the change in this PR. It is definitively a better approach. Thanks :)